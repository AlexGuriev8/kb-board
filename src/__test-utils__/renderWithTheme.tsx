/* eslint-disable import/no-extraneous-dependencies */
import { ReactElement, ReactNode } from 'react';
import { render as customRender, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import { StoreProvider } from '../store/createStoreContext';

interface IProps {
  children: ReactNode;
}

export type IRendersOptions = RenderOptions;

export const mockTheme = {
  colors: {
    primary: '#fff',
    backgroundColor: '#2b2c37',
    borderRightColor: '#3e3f4ed9',
    switchBgColor: '#20212C',
    contentInfo: '#20212C',
    modalColor: '#2B2C37',
    backdrop: 'rgba(0, 0, 0, 0.2)',
    heading: '#fff',
    placeholder: '##D0D0D0',
    hovers: {
      danger: '',
    },
  },
};

function render(
  component: ReactElement,
  { ...renderOptions }: IRendersOptions = {}
) {
  const Wrapper = ({ children }: IProps) => {
    return (
      <StoreProvider>
        <ThemeProvider theme={mockTheme}>{children}</ThemeProvider>
      </StoreProvider>
    );
  };
  return customRender(component, { wrapper: Wrapper, ...renderOptions });
}
export { render };
