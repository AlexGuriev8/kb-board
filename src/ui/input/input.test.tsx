import React from 'react';
import { screen } from '@testing-library/react';
import Input from '.';
import { render } from '../../__test-utils__/renderWithTheme';

describe('Input', () => {
  it('should render the component correctly', () => {
    const { container } = render(
      <Input
        name="test"
        label="Test Input"
        value="Test Value"
        onChange={() => {}}
        placeholder="Test Placeholder"
        className="test-class"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the input field with the correct props', () => {
    render(
      <Input
        name="test"
        label="Test Input"
        value="Test Value"
        onChange={() => {}}
        placeholder="Test Placeholder"
        className="test-class"
      />
    );
    const input = screen.getByTestId('input-element');

    expect(input).toHaveAttribute('id', 'test');
    expect(input).toHaveAttribute('value', 'Test Value');
    expect(input).toHaveAttribute('placeholder', 'Test Placeholder');
    expect(input).toHaveAttribute('name', 'test');
  });
});
