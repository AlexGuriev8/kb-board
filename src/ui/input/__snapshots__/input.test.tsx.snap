// Vitest Snapshot v1

exports[`Input > should render the component correctly 1`] = `
<div>
  <div
    class="test-class css-5zdpsq"
  >
    <label
      for="test"
    >
      Test Input
    </label>
    <div
      class="input-box"
    >
      <input
        data-testid="input-element"
        id="test"
        name="test"
        placeholder="Test Placeholder"
        type="text"
        value="Test Value"
      />
    </div>
  </div>
</div>
`;
